apps/sms/views/shared/js/settings.js

```diff
@@ -72,6 +79,15 @@ var Settings = {
     }
   },
 
+  // return resolve promise with sendReadReport status
+  initSendReadReport: function initSendReadReport(value) {
+    this._sendReadReportDefer.resolve(value);
+    delete this._sendReadReportDefer;
+
+    navigator.mozSettings.addObserver('messages.mms.sendReadReport.enabled',
+      (e) => this.sendReadReport = Promise.resolve(e.settingValue));
```

Comment:
@steveck-chung  I asked same from julienw, Actually, it possible that on startup the value is not set. But that's also true that, init() runs first before any method/navigation do. (and all are navigation are promise based)
Or i guess @julienw  can explain better.